src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -45,33 +60,211 @@ static NSStringCompareOptions ConvertFromCompareOptionsToNSStringCompareOptions(
 Function:
 CompareString
 */
-int32_t GlobalizationNative_CompareStringNative(const uint16_t* localeName, int32_t lNameLength, const uint16_t* lpStr1, int32_t cwStr1Length, 
-                                                const uint16_t* lpStr2, int32_t cwStr2Length, int32_t comparisonOptions)
+int32_t GlobalizationNative_CompareStringNative(const uint16_t* localeName, int32_t lNameLength, const uint16_t* lpSource, int32_t cwSourceLength, 
```

Comment:
The reason we normalize here is because we don't care about the length we just need to check if they are equal strings, so using any normalization form should work here. 
In the documentation it is not specified that compare is using canonical equivalence so my guess is all forms are supported here.